spec/lib/metasploit/framework/credential_collection_spec.rb

```diff
@@ -4,6 +4,10 @@
 RSpec.describe Metasploit::Framework::CredentialCollection do
 
   subject(:collection) do
+    allow(File).to receive(:open).with(anything).and_call_original
+    allow(File).to receive(:open).with(anything, anything).and_call_original
+    allow(File).to receive(:open).with(anything, anything, anything).and_call_original
```

Comment:
Might be good to rip this out of the `subject`, and move it into its own `before` block

```ruby
  before(:each) do
    # The test suite overrides File.open(...) calls; fall back to the normal behavior for any File.open calls that aren't explicitly mocked
    allow(File).to receive(:open).with(anything).and_call_original
    allow(File).to receive(:open).with(anything, anything).and_call_original
    allow(File).to receive(:open).with(anything, anything, anything).and_call_original
  end
```